compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Validate.hs

```diff
@@ -0,0 +1,71 @@
+-- Copyright (c) 2020 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.Compiler.Validate (validateDar) where
+
+import Control.Monad (forM_)
+import qualified "zip-archive" Codec.Archive.Zip as ZipArchive
+import qualified DA.Daml.LF.Ast as LF
+import qualified DA.Daml.LF.Proto3.Archive as Archive
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.NameMap as NM
+
+import DA.Daml.Compiler.ExtractDar (extractDar,ExtractedDar(..))
+import DA.Daml.LF.Ast.Version
+import DA.Daml.LF.Ast.World (initWorld)
+import DA.Pretty (renderPretty)
+import qualified DA.Daml.LF.TypeChecker as TC (checkModule)
+import qualified DA.Daml.LF.TypeChecker.Error as TC
+
+-- | Monad for performing Validation
+newtype Validation a = Validation (Either ValidationError a) deriving (Functor,Applicative,Monad)
```

Comment:
Given that this type is only used internally in this module, I would just use  `Either ValidationError` directly. That also avoids confusion with the relatively popular `Validation` type in Haskell.

Reply:
I removed the `Validation` type entirely. It was buying almost nothing.